src/v/redpanda/admin_server.cc

```diff
@@ -1541,6 +1541,41 @@ void admin_server::register_features_routes() {
           }
           co_return std::move(res);
       });
+
+    register_route<superuser>(
+      ss::httpd::features_json::put_license,
+      [this](std::unique_ptr<ss::httpd::request> req)
+        -> ss::future<ss::json::json_return_type> {
+          auto& raw_license = req->content;
+          if (raw_license.empty()) {
+              throw ss::httpd::bad_request_exception(
+                "Missing redpanda license from request body");
+          }
+          try {
+              auto license = security::make_license(raw_license);
+              if (license.is_expired()) {
+                  throw ss::httpd::bad_request_exception(fmt::format(
+                    "Expired license sent to redpanda: {}", license));
+              }
+              auto& fm = _controller->get_feature_manager();
+              co_await fm.invoke_on(
+                cluster::feature_manager::backend_shard,
+                [license = std::move(license)](
+                  cluster::feature_manager& fm) mutable {
+                    return fm.update_license(std::move(license));
+                });
+          } catch (const security::license_malformed_exception& ex) {
+              throw ss::httpd::bad_request_exception(fmt::format(
+                "Malformed license sent to redpanda: {}", ex.what()));
+          } catch (const security::license_invalid_exception& ex) {
+              throw ss::httpd::bad_request_exception(
+                fmt::format("Invalid license sent to redpanda: {}", ex.what()));
+          } catch (const std::runtime_error& ex) {
```

Comment:
In the case the feature manager fails to disseminate the license. Most likely because the node is not the leader.

Reply:
Oh, now I see where runtime_error is being raised in feature_manager::update_license.

I think this should change to have update_license return an error code (which might be errc::not_leader), and then call `admin_server::throw_on_error` on the result -- that way this code will be clearer, and clients calling on the non-leader will get a redirect to the leader courtesy of the existing logic in `throw_on_error`.

This will also enable simplifying the `put_license` python function in `Admin` to skip looking up the leader.